src/main/java/wooteco/subway/domain/Section.java

```diff
@@ -0,0 +1,135 @@
+package wooteco.subway.domain;
+
+import java.util.Objects;
+
+public class Section {
+
+    private static final int MIN_DISTANCE = 1;
+    private static final String INVALID_DISTANCE_ERROR_MESSAGE = String.format("거리는 %d 이상이어야 합니다.", MIN_DISTANCE);
+    private static final String DUPLICATED_SECTIONS_ERROR_MESSAGE = "상행과 하행은 같은 역으로 등록할 수 없습니다.";
+
+    private Long id;
+    private Long lineId;
+    private Long upStationId;
+    private Long downStationId;
+    private int distance;
+
+    public Section(Long lineId, Long upStationId, Long downStationId, int distance) {
+        validSection(upStationId, downStationId, distance);
+        this.lineId = lineId;
+        this.upStationId = upStationId;
+        this.downStationId = downStationId;
+        this.distance = distance;
+    }
+
+    public Section(Long id, Long lineId, Long upStationId, Long downStationId, int distance) {
+        this(lineId, upStationId, downStationId, distance);
+        this.id = id;
+    }
```

Comment:
주생성자에서 유효성 검사와 모든 필드를 초기화하도록 수정하면 어떨까요?
지금 상태에서는 부생성자가 여러개 생긴다면 각 생성자들이 도메인 규칙을 잘 지켰는지 파악하기 어려울 것 같아요.

```suggestion
    public Section(Long lineId, Long upStationId, Long downStationId, int distance) {
        this(null, lineId, upStationId, downStationId, distance);
    }

    public Section(Long id, Long lineId, Long upStationId, Long downStationId, int distance) {
        validSection(upStationId, downStationId, distance);
        this.id = id;
        this.lineId = lineId;
        this.upStationId = upStationId;
        this.downStationId = downStationId;
        this.distance = distance;
    }
```